README.md

```diff
@@ -1,17 +1,95 @@
+It captures relationships between accounts using a _directed graph_, where nodes are Sui addresses and edges are relationships between a pair of Sui addresses.
 It also provides the ability to store properties of nodes and edges along with the graph.
 All graph operations are authorized by the source account.
 
-## Build
-`sui move build`
+Many account graph instances can be created, each capturing one kind of account relationship, with different properties and constraints.
 
-## Test
-`sui move test`
+**For information regarding Bullshark Quests integration, see [Bullshark Quests beneficiary graph](#bullshark-quests-beneficiary-graph) and [Deployed resources](#deployed-resources).**
+
+## Graph constraints
+
+Currently, the only supported graph constraint is `max_out_degree`, which limits how many outgoing relationships a source account can have.
+The constraint is specified during graph instantiation, and enforced during all later graph operations.
+
+## Graph operations
+
+Each account graph instance is a shared Move object, so anyone can interact with it.
+However, all graph operations are implicitly performed against the transaction `sender` account.
+
+### Manage relationships
+
+An account can manage its own outgoing relationships in a graph through these move calls:
+
+- `<ACCOUNT_GRAPH_PKG>::account_graph::add_relationship` - adds a relationship from the transaction `sender` account to an arbitrary target account.
+- `<ACCOUNT_GRAPH_PKG>::account_graph::remove_relationship` - removes a relationship from the transaction `sender` account to an arbitrary target account.
+- `<ACCOUNT_GRAPH_PKG>::account_graph::clear_relationship` - clears all relationships from the transaction `sender` account.
+
+### Manage account properties
+
+An account can optionally manage its own account properties in a graph.
+The type of account properties is specified by the graph creator during instantiation.
+
+- `<ACCOUNT_GRAPH_PKG>::account_graph::set_account_props` - sets account properties for the transaction `sender` account.
+- `<ACCOUNT_GRAPH_PKG>::account_graph::unset_account_props` - unsets account properties for the transaction `sender` account.
+
+### Manage relationship properties
+
+An account can optionally manage properties of its outgoing relationships in a graph.
+The type of relationship properties is specified by the graph creator during instantiation.
+
+- `<ACCOUNT_GRAPH_PKG>::account_graph::set_relationship_props` - sets properties for the relationship from the transaction `sender` account to a target account.
+- `<ACCOUNT_GRAPH_PKG>::account_graph::unset_relationship_props` - unsets properties for the relationship from the transaction `sender` account to a target account.
+
+## Bullshark Quests beneficiary graph
+
+The account graph has been adopted by the [Bullshark Quests](https://quests.mystenlabs.com/) as the official way to link a user's managed in-app wallet to their self-custody wallet.
```

Comment:
we should put whichever of these is correct:

"...as the official way to link..."
"...as the official and only way to link..."